bin/smoke

```diff
@@ -47,18 +47,26 @@ class NonExecutableApplicationError < SmokeError
   end
 end
 
-def validate_application(application)
-  raise NoSuchApplicationError, application unless File.exist?(application)
-  raise NonExecutableApplicationError, application unless File.executable?(application)
-end
```

Comment:
Also the change invalidates the readme. Now it is not possible to provide commands like ruby script.rb any more. But that is not a problem in Linux of course.